crates/typst/src/loading/json.rs

```diff
@@ -9,13 +9,18 @@ use crate::World;
 
 /// Reads structured data from a JSON file.
 ///
-/// The file must contain a valid JSON object or array. JSON objects will be
-/// converted into Typst dictionaries, and JSON arrays will be converted into
-/// Typst arrays. Strings and booleans will be converted into the Typst
-/// equivalents, `null` will be converted into `{none}`, and numbers will be
-/// converted to floats or integers depending on whether they are whole numbers.
+/// The file must contain a valid JSON value, such as object or array. JSON
+/// objects will be converted into Typst dictionaries, and JSON arrays will be
+/// converted into Typst arrays. Strings and booleans will be converted into the
+/// Typst equivalents, `null` will be converted into `{none}`, and numbers will
+/// be converted to floats or integers depending on whether they are whole
+/// numbers.
 ///
-/// The function returns a dictionary or an array, depending on the JSON file.
+/// Be aware that integers larger than 2<sup>63</sup>-1 will be converted to
+/// floating point numbers, which may give an approximative value.
```

Comment:
This is still on topic for the PR, so we can just rename it and broaden the scope a bit

Reply:
Okay! For info:
- CBOR and YAML have the same issue as JSON. I will update the docs for these functions as well.
- CSV and XML consider every primitive value to be a string, so this issue is not relevant
- TOML reports `Failed to parse TOML (number too large to fit in target type at line 1 column 10)` which comes from the `toml` crate directly I believe. Ideally this should be made consistent with JSON, CBOR and YAML, but I think it is not such a big issue, and fixing this would probably be a bit more involved and out of the scope of this PR.

Reply:
Upstream TOML bug: https://github.com/toml-rs/toml/issues/705

Reply:
Sounds good